src/librustc_mir/transform/instcombine.rs

```diff
@@ -11,44 +11,54 @@
 //! Performs various peephole optimizations.
 
```

Comment:
cc @nikomatsakis @scottmcm This file ended up with a constant folder, can you take a look?